xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/query/hql/internal/HQLCompleteStatementValidator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.query.hql.internal;
+
+import java.util.Optional;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.query.QueryException;
+
+/**
+ * A component in charge of validating a passed HQL statement.
```

Comment:
Would it make sense to document the use of priorities here and the need to have a priority below the default for a validator to be used?

Reply:
You are right, it would indeed make sense to indicate that a validator won't be used if it does not have a priority lower than the standard one.